homeassistant/components/climate/ecobee.py

```diff
@@ -382,13 +383,3 @@ def hold_preference(self):
             return default
         else:
             return 'nextTransition'
-
-    # Sleep mode isn't used in UI yet:
```

Comment:
FYI, Sleep mode can be a hold. Any climate (even ones you define) in ecobee can be a hold. See https://github.com/home-assistant/home-assistant/pull/5590

Now, does that mean we should keep this code in here? probably not, since it isn't used. I personally think that the toggle switch for "home" or "away" mode is the wrong choice and all of these functions should go away. I think we should move to an input select which would automatically be populated with home, away, temp, and sleep (well, each thermostat would have to provide their own list), and be extendable to add smart1, through smartX as the user needs, and it would be even better if the user can map meaningful names to the smartX climates.